libraries/xcode.rb

```diff
@@ -1,24 +1,54 @@
 include Chef::Mixin::ShellOut
 
 module MacOS
-  module Xcode
-    class << self
-      def installed?(semantic_version)
-        xcversion_output = shell_out(XCVersion.installed_xcodes).stdout.split
-        installed_xcodes = xcversion_output.values_at(*xcversion_output.each_index.select(&:even?))
-        installed_xcodes.include?(semantic_version)
-      end
+  class Xcode
+    attr_reader :version
+    attr_reader :credentials
 
-      def find_apple_id(data_bag_retrieval, node_credential_attributes)
-        if node_credential_attributes
-          { 'apple_id' => node_credential_attributes['user'],
-            'password' => node_credential_attributes['password'] }
-        else
-          data_bag_retrieval.call
-        end
-      rescue Net::HTTPServerException
-        Chef::Application.fatal!('No developer credentials supplied!')
+    def initialize(semantic_version, data_bag_retrieval = nil, node_credential_attributes = nil)
+      @semantic_version = semantic_version
+      developer_id = find_apple_id(data_bag_retrieval, node_credential_attributes)
+      @credentials = {
+        XCODE_INSTALL_USER:     developer_id['apple_id'],
+        XCODE_INSTALL_PASSWORD: developer_id['password'],
+      }
+      authenticate_with_apple(@credentials)
+      @version = apple_pseudosemantic_version(semantic_version)
+    end
+
+    def apple_pseudosemantic_version(semantic_version)
```

Comment:
This method seems to be doing a lot. Furthermore, it looks like it's actually returning available Xcode versions that match the passed in version? I think? It's a bit hard to follow.

Reply:
Ah, it's returning the first version that matches the passed in version. Either way, this method is probably doing too much.

Reply:
Thanks, I've split this up.